yeat/config/assembly.py

```diff
@@ -55,3 +62,47 @@ def check_canu_required_params(self):
         if self.threads < 4:
             message = f"Canu requires at least 4 avaliable cores; increase '-t' or '--threads' to 4 or more"
             raise AssemblyConfigError(message)
+
+    def check_sample_readtypes_match_assembly_mode(self):
+        for sample in self.samples.values():
+            check = self.is_mode_and_sample_readtypes(sample)
+            self.check_sample_readtypes_match(check, sample)
+
+    def is_mode_and_sample_readtypes(self, sample):
+        if self.mode == "paired" and sample.short_readtype != "paired":
+            return False
+        elif self.mode == "single" and sample.short_readtype != "single":
+            return False
+        elif self.mode == "pacbio" and sample.long_readtype not in PACBIO_READS:
+            return False
+        elif self.mode == "oxford" and sample.long_readtype not in OXFORD_READS:
+            return False
+        elif self.mode == "hybrid" and (
+            sample.short_readtype != "paired" or sample.long_readtype not in LONG_READS
+        ):  # pragma: no cover
+            return False
+        return True
+
+    def check_sample_readtypes_match(self, check, sample):
+        if check == False:
+            message = f"No readtypes in '{sample.label}' match '{self.label}' assembly mode '{self.mode}'"
+            raise AssemblyConfigError(message)
+
+    def get_expected_files(self):
+        expected_files = []
+        for sample in self.samples.values():
+            expected_files.append(self.get_qa_file(sample))
+        return expected_files
+
+    def get_qa_file(self, sample):
+        readtype = self.get_readtype(sample)
+        algorithm_dir = f"analysis/{sample.label}/{readtype}/{self.label}/{self.algorithm}"
+        if self.bandage:
+            return f"{algorithm_dir}/bandage/.done"
+        return f"{algorithm_dir}/quast/report.html"
+
+    def get_readtype(self, sample):
+        if self.mode in ["paired", "single", "hybrid"]:
+            return self.mode
+        elif self.mode in ["pacbio", "oxford"]:  # pragma: no cover
+            return sample.long_readtype
```

Comment:
These three functions calculates the final expected file for each sample in the assembly's directory.

For example,
Assembly object has 3 samples: ["sample1", "sample2", and "sample3"].
User's computer does not have bandage.
Expected final files are:
- analysis/**sample1**/paired/default-spades/spades/quast/report.html
- analysis/**sample2**/paired/default-spades/spades/quast/report.html
- analysis/**sample3**/paired/default-spades/spades/quast/report.html